CMakeLists.txt

```diff
@@ -197,6 +197,18 @@ if(CONFIG_MISRA_SANE)
   zephyr_compile_options($<$<COMPILE_LANGUAGE:CXX>:${CPP_MISRA_SANE_FLAG}>)
 endif()
 
+if(CONFIG_NEWLIB_LIBC)
+# @Intent: Enforce standard type correspondance to match Zephyr usage.
+# See lib/libc/minimal/include/stdint.h for details.
+# (note: zephyr_compile_options() doesn't work with "SHELL:")
+add_compile_options(
+	-U__INT32_TYPE__	"SHELL:-D__INT32_TYPE__='int'"
+	-U__UINT32_TYPE__	"SHELL:-D__UINT32_TYPE__='unsigned int'"
+	-U__INTPTR_TYPE__	"SHELL:-D__INTPTR_TYPE__='long int'"
+	-U__UINTPTR_TYPE__	"SHELL:-D__UINTPTR_TYPE__='long unsigned int'"
+)
```

Comment:
What do you think about using the gcc option

-imacros enforce_standard_type_correspondance.h

instead here?

This would shorten the command line and allow us to use

#undef

and

#define

which readers will be more familiar with.

Also allows us to avoid using SHELL, which I am not familiar with, and add_compile_options, which causes problems for the zephyr_get_* API.

imacro should generally be avoided, as symbols should not magically appear in the source, but symbols that are prefixed with __ shouldn't be expected to be in-source anyway.